src/Build/NuGet.Build.Packaging.Tests/CreatePackageTests.cs

```diff
@@ -149,6 +149,56 @@ public void when_creating_package_with_simple_dependency_then_contains_dependenc
 			Assert.Equal("8.0.0", manifest.Metadata.DependencyGroups.First().Packages.First().VersionRange.MinVersion.ToString());
 		}
 
+		[Fact]
+		public void when_creating_package_with_dependency_and_include_assets_then_contains_dependency_include_attribute()
+		{
+			task.Contents = new[]
+			{
+				new TaskItem("Newtonsoft.Json", new Metadata
+				{
+					{ MetadataName.PackageId, task.Manifest.GetMetadata("Id") },
+					{ MetadataName.Kind, PackageItemKind.Dependency },
+					{ MetadataName.Version, "8.0.0" },
+					// NOTE: AssignPackagePath takes care of converting TFM > short name
+					{ MetadataName.TargetFramework, "net45" },
+					{ MetadataName.IncludeAssets, "all" }
```

Comment:
This should be quite useful indeed. I'd add support for `ExcludeAssets` too, to complete the feature.